is_core/utils/__init__.py

```diff
@@ -87,200 +73,136 @@ def get_inline_views_opts_from_fieldsets(fieldsets):
 
 
 def get_field_from_model_or_none(model, field_name):
+    """
+    Return field from model. If field doesn't exists null is returned instead of exception.
+    """
     try:
         return model._meta.get_field(field_name)
     except (FieldDoesNotExist, AttributeError):
         return None
 
 
-def _get_verbose_value(raw, field_or_method, obj, **kwargs):
-    if hasattr(field_or_method, 'humanized') and field_or_method.humanized:
-        return field_or_method.humanized(raw, obj, **kwargs)
-    elif hasattr(field_or_method, 'attname') and hasattr(obj, 'get_{}_display'.format(field_or_method.attname)):
-        return getattr(obj, 'get_{}_display'.format(field_or_method.attname))()
-    else:
-        return raw
-
-
-def _val_to_readonly_value(value, field_or_method, obj, **kwargs):
-    from is_core.forms.utils import ReadonlyValue
-    verbose_value = _get_verbose_value(value, field_or_method, obj, **kwargs)
-    return ReadonlyValue(value, verbose_value) if value != verbose_value else value
-
-
-def _get_method_value(method, field_name, inst, fun_kwargs):
-    method_kwargs_names = inspect.getfullargspec(method)[0][1:]
-    method_kwargs = {arg_name: fun_kwargs[arg_name]  for arg_name in method_kwargs_names if arg_name in fun_kwargs}
-    if len(method_kwargs_names) == len(method_kwargs):
-        return _val_to_readonly_value(method(**method_kwargs), method, inst,
-                                     **{k: v for k, v in method_kwargs.items() if k != 'obj'})
-    else:
-        raise(InvalidMethodArguments('Method {} arguments has not subset of fun kwargs'.format(field_name)))
+def get_field_label_from_path(model, field_path, view=None, field_labels=None):
+    """
+    Return field label of model class for input field path. For every field name in the field path is firstly get the
+    right label and these labels are joined with " - " separator to one string.
+
+    field_label input parameter can affect the result value. Example:
+
+    * field_path='user__email', field_labels={} => 'user email'  # default values get from fields
+    * field_path='user__email', field_labels={'user__email': 'e-mail'} => 'e-mail'  # full value is replaced
+    * field_path='user__email', field_labels={'user': 'customer'} => 'customer email'  # related field prefix is changed
```

Comment:
Shouldn't this be `customer - email`? The docstring and code states, that names are joined with ` - `.

Reply:
yes thanks